databuilder/query_language.py

```diff
@@ -303,6 +303,16 @@ def is_after(self, other):
     def is_on_or_after(self, other):
         return self.__ge__(other)
 
+    def is_in(self, other):
+        # The query model requires an immutable Set type for containment queries, but
+        # that's a bit user-unfriendly so we accept other types here and convert them
+        if isinstance(other, (tuple, list, set)):
+            # Parse any ISO date strings in collection
+            t = type(other)
+            other = t(map(parse_date_if_str, other))
+            other = frozenset(other)
+        return _apply(qm.Function.In, self, other)
```

Comment:
Coverage is moaning at you because you've not tested what happens if this condition is false (i.e. you pass something which isn't a tuple, list or set). But I think that exposes an issue here which is that you want to apply the date parsing to _any_ iterable that's passed in, not just those which need converting to `frozenset`. So given that I think you could throw away the conditional all together, which then gives you fewer branches to test. Something like:
```py
# Parse any ISO date strings in collection
other = frozenset(map(parse_date_if_str, other))
return _apply(qm.Function.In, self, other)
```

Reply:
I guess on this path it would never get given a Codelist which was what was tripping me up last time, I'll remove the conditional and see what happens.

Reply:
`just test-all` seems happier locally with that